Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -35,1657 +35,1664 @@
+                    {
+                        decltype(local_placement->RelativePlacement)& axis2placement = local_placement->RelativePlacement;
+                        switch(axis2placement.which()) {
+                        case 0:
+                            convertIfcAxis2Placement2D(axis2placement.get<0>().lock(), object_placement_matrix);
+                            break;
+                        case 1:
+                            convertIfcAxis2Placement3D(axis2placement.get<1>().lock(), object_placement_matrix);
+                            break;
+                        default:
+                            BLUE_LOG(fatal) << local_placement->getErrorLog() << "RelativePlacement conversion issues.";
+                            break;
+                        }
+                    }
+
+                    void convertIfcLocalPlacementRelationPoint(std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcObjectPlacement>>& alreadyApplied, std::shared_ptr<typename IfcEntityTypesT::IfcLocalPlacement> local_placement, carve::math::Matrix &relative_placement)
+                    {
+                        // PlacementRelTo
+                        if(local_placement->PlacementRelTo) {
+                            // Reference to ObjectPlacement that provides the relative placement by its local coordinate system. 
+                            convertIfcObjectPlacement(local_placement->PlacementRelTo.get().lock(), relative_placement, alreadyApplied);
+                        }
+                        else {
+                            BLUE_LOG(warning) << "Context based local placement computation not supported.";
+                            //TODO Georeferencing
+                            // If reference to Object is omitted, then the local placement is given to the WCS, established by the geometric representation context.
+                            //carve::math::Matrix context_matrix( carve::math::Matrix::IDENT() );
+                            //applyContext( context, context_matrix, length_factor, placement_already_applied );
+                            //object_placement_matrix = context_matrix*object_placement_matrix;
+                        }
+                    }
+
+
+                    /*! \brief Converts \c IfcLocalPlacement to a transformation matrix.
+
+                    \param	alreadyApplied		An array of references to already applied \c IfcObjectPlacement-s.
+                    \param[in]	local_placement		\c IfcLocalPlacement entity to be interpreted.
+                    \param[out] matrix			Calculated transformation matrix.
+
+                    Function computes the local placement and stores it in the matrix.
+                    */
+                    void convertIfcLocalPlacement(std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcObjectPlacement>>& alreadyApplied, carve::math::Matrix& object_placement_matrix, std::shared_ptr<typename IfcEntityTypesT::IfcLocalPlacement> local_placement)

```

Comment:
Change the [out] parameter to a return value.

Reply:
Done